lib/kaur/result.ex

```diff
@@ -91,6 +91,52 @@ defmodule Kaur.Result do
   def from_value(value), do: ok(value)
 
   @doc """
+  Converts an `Ok` value to an `Error` value if the `predicate` is not valid.
+
+  ## Examples
+
+      iex> res = Kaur.Result.ok(10)
+      ...> Kaur.Result.keep_if(res, &(&1 > 5))
+      {:ok, 10}
+
+      iex> res = Kaur.Result.ok(10)
+      ...> Kaur.Result.keep_if(res, &(&1 > 10), "must be > of 10")
+      {:error, "must be > of 10"}
+
+      iex> res = Kaur.Result.error(:no_value)
+      ...> Kaur.Result.keep_if(res, &(&1 > 10), "must be > of 10")
+      {:error, :no_value}
+  """
+  @spec keep_if(result_tuple, (any -> boolean), any) :: result_tuple
+  def keep_if(result, predicate, error_message \\ :invalid)
```

Comment:
I don't understand when it's going to be useful to have this default error message?

Reply:
When you use it without any interest for an error message.
That was the use case of @arnaudmorisset